lib/chat/bloc/provider/ChatProvider.dart

```diff
@@ -0,0 +1,307 @@
+      _insertProvidedMessage(providedMessage);
+    } else {
+      _insertMessageFromService();
+    }
+    _scrollToBottom();
+  }
+
+  void _insertProvidedMessage(String providedMessage) {
+    _insertNewMessageToList(
+        _chatMessageHandler.getMessageFromString(providedMessage));
+    _notifyController();
+    _showMessageWithDelay();
+  }
+
+  void _insertLoadingMessage() {
+    _insertNewMessageToList(_chatMessageHandler.getLoadingMessage());
+    _notifyController();
+  }
+
+  void _removeLoadingMessage() {
+    _chatViewModel.messageViewModels.removeAt(_Constants.currentMessageIndex);
+    _notifyController();
+  }
+
+  void _insertMessageFromService() {
+    _repo.getFormItem(_currentMessageCount).then((formItem) {
+      if (formItem != null) {
+        _insertNewMessageToList(
+            _chatMessageHandler.getMessageFromEntity(formItem));
+        _increaseMessageCount();
+        _updateAvatarVisibility();
+        _setInteractiveWidget(formItem.inputRequest);
+        _notifyController();
+      } else {
+        _setSummaryWidget();
+      }
+    });
+  }
+
+  void _setSummaryWidget() {
+    _chatViewModel.interactiveWidget = CardView(
+      child: Summary(
+        viewModel: _chatSummaryProviderHelper.getSummary(
+          inputModel: _responseMap,
+          titleValue: _Strings.summaryTitleValue,
+          titleText: _Strings.summaryTitleLabel.toUpperCase(),
+          actionText: _Strings.summaryActionButtonText,
+        ),
+        onTap: _onSummaryWidgetActionButtonTap,
+      ),
+    );
+  }
+
+  void _onSummaryWidgetActionButtonTap() {
+    (_responseMap != null && _responseMap.isNotEmpty)
+        ? _onSuccess(_responseMap)
+        : _onError(_Strings.summaryError);
+  }
+
+  void _setInteractiveWidget(InputRequestEntity entity) {
+    if (entity != null && entity.type != null) {
+      switch (entity.type) {
+        case _Constants.typeValueString:
+          _chatViewModel.interactiveWidget = _buildInputTextWidget(
+            entity: entity,
+            inputType: InteractiveMessageType.inputString,
+          );
+          break;
+        case _Constants.typeValueEmail:
+          _chatViewModel.interactiveWidget = _buildInputTextWidget(
+            entity: entity,
+            inputType: InteractiveMessageType.inputEmail,
+          );
+          break;
+        case _Constants.typeValuePhoneNumber:
+          _chatViewModel.interactiveWidget = _buildInputTextWidget(
+            entity: entity,
+            inputType: InteractiveMessageType.inputPhoneNumber,
+          );
+          break;
+        case _Constants.typeValueImage:
+          _chatViewModel.interactiveWidget = _buildInputTextWidget(
+            entity: entity,
+            inputType: InteractiveMessageType.inputImage,
+          );
+          break;
+        case _Constants.typeValueMultiChoice:
+          _chatViewModel.interactiveWidget =
+              _buildSelectableOptionsWidget(entity: entity);
+          break;
+      }
```

Comment:
I think that this should be the same:

```
      InteractiveMessageType inputType;
      switch (entity.type) {
        case _Constants.typeValueString:
          inputType = InteractiveMessageType.inputString;
          break;
        case _Constants.typeValueEmail:
          inputType = InteractiveMessageType.inputEmail;
          break;
        case _Constants.typeValuePhoneNumber:
          inputType = InteractiveMessageType.inputPhoneNumber;
          break;
        case _Constants.typeValueImage:
          inputType = InteractiveMessageType.inputImage;
          break;
        case _Constants.typeValueMultiChoice:
          inputType = null;
          break;
      }

      _chatViewModel.interactiveWidget = _buildInputTextWidget(
        entity: entity,
        inputType: inputType,
      );
```

Reply:
It would be nice if we can have enums with values, so then we would reduce this switch, not sure if we can do it with flutter.

Reply:
Refactored into 2 functions to make it more readable.
Do not see the need of having the value for enum, I just need to know the type of provided input. Could you provide some additional info?

Reply:
👍